drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -365,8 +365,12 @@ NTSTATUS TCPConnect
         return Status;
     }
 
-    /* Copy bind address into connection */
-    Connection->AddressFile->Address.Address.IPv4Address = bindaddr.addr;
+    /* Check if we had an unspecified address */
+    if (Connection->AddressFile->Address.Address.IPv4Address != bindaddr.addr)
+    {
+         /* We did, so we need to copy back the address */
+         Connection->AddressFile->Address.Address.IPv4Address = bindaddr.addr;
+    }
```

Comment:
Why can't we just always copy back the address?

Reply:
Only change I made was commit [f5f4740](https://github.com/reactos/reactos/pull/5092/commits/f5f47402da15870c579a02c3dc8eea3c0d4987f2)
Sorry I am trying to clean up the commit history right now. The other changes were made by hater.

Reply:
Commit history is fixed, the first commit [0d9ab98](https://github.com/reactos/reactos/pull/5092/commits/0d9ab989975193f597dace30610217503684adcf) and mine is [f5f4740](https://github.com/reactos/reactos/pull/5092/commits/f5f47402da15870c579a02c3dc8eea3c0d4987f2)